File: 250331/흥미로운 숫자 2/interesting-numbers-2.c
#include <stdio.h>
#include <string.h>

int is_interesting(int num) {
    int freq[10] = {0};
    char str[12];  // int 범위 내 숫자를 문자열로 바꿔 저장
    sprintf(str, "%d", num);

    int len = strlen(str);

    // 각 숫자 빈도수 계산
    for (int i = 0; i < len; i++) {
        int digit = str[i] - '0';
        freq[digit]++;
    }

    int unique_digits = 0;
    int counts[2]; // 등장하는 두 숫자의 빈도 저장
    int idx = 0;

    for (int i = 0; i < 10; i++) {
        if (freq[i] > 0) {
            unique_digits++;
            if (idx < 2) {
                counts[idx++] = freq[i];
            }
        }
    }

    if (unique_digits != 2) {
        return 0;
    }

    // 정렬 없이 조건 확인
    int a = counts[0], b = counts[1];
    if ((a == 1 && b == len - 1) || (b == 1 && a == len - 1)) {
        return 1;
    }

    return 0;
}

int count_interesting_numbers(int X, int Y) {
    int count = 0;
    for (int num = X; num <= Y; num++) {
        if (is_interesting(num)) {
            count++;
        }
    }
    return count;
}

int main() {
    int X, Y;
    scanf("%d %d", &X, &Y);

    int result = count_interesting_numbers(X, Y);
    printf("%d\n", result);

    return 0;
}

File: 250331/흥미로운 숫자 2/interesting-numbers-2.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int X = sc.nextInt();
        int Y = sc.nextInt();
        
        int count = 0;
        for (int num = X; num <= Y; num++) {
            if (is_interesting(num)) {
                count++;
            }
        }

        System.out.printf("%d", count);
    }

    public static boolean is_interesting(int num) {
        int[] freq = new int[10];
        int len = 0;
        int temp = num;

        while (temp > 0) {
            int digit = temp % 10;
            freq[digit]++;
            len++;
            temp /= 10;
        }

        if (len < 2) return false;

        int uniqueDigits = 0;
        int[] counts = new int[2];
        int idx = 0;

        for (int i = 0; i < 10; i++) {
            if (freq[i] > 0) {
                uniqueDigits++;
                if (idx < 2) {
                    counts[idx++] = freq[i];
                }
            }
        }

        if (uniqueDigits != 2) return false;

        int a = counts[0], b = counts[1];
        return ((a == 1 && b == len - 1) || (b == 1 && a == len - 1));
    }
}